cmd/ruler/main.go

```diff
@@ -59,28 +58,29 @@ func main() {
 	util.CheckFatal("", err)
 	defer chunkStore.Stop()
 
-	r, err := ring.New(ringConfig)
+	r, err := ring.New(lifecyclerConfig.RingConfig)
 	util.CheckFatal("initializing ring", err)
+
 	prometheus.MustRegister(r)
 	defer r.Stop()
 
 	dist, err := distributor.New(distributorConfig, clientConfig, overrides, r)
 	util.CheckFatal("initializing distributor", err)
 	defer dist.Stop()
 
+	ruleStore, err := ruler.NewRuleStore(ruleStoreConfig)
+	util.CheckFatal("error initializing rules API", err)
+
 	querierConfig.MaxConcurrent = rulerConfig.NumWorkers
 	querierConfig.Timeout = rulerConfig.GroupTimeout
 	queryable, engine := querier.New(querierConfig, dist, chunkStore)
-	rlr, err := ruler.NewRuler(rulerConfig, engine, queryable, dist)
-	util.CheckFatal("initializing ruler", err)
-	defer rlr.Stop()
 
-	rulesAPI, err := ruler.NewRulesAPI(configStoreConfig)
-	util.CheckFatal("initializing rules API", err)
-
-	rulerServer, err := ruler.NewServer(rulerConfig, rlr, rulesAPI)
-	util.CheckFatal("initializing ruler server", err)
-	defer rulerServer.Stop()
+	rulerConfig.LifecyclerConfig = lifecyclerConfig
```

Comment:
Can we have the rulerConfig recursively register the lifecycler, so we don't have to do this?  This has been error prone in the past.